app/_src/gateway/kong-enterprise/fips/fips-support.md

```diff
@@ -0,0 +1,54 @@
+---
+title: About FIPS 140-2 Compliance in Kong Gateway
+badge: enterprise
+content_type: reference
+---
+
+The Federal Information Processing Standard (FIPS) 140-2 is a federal standard defined by the National Institute of Standards and Technology. It specifies the security requirements that must be satisfied by a cryptographic module. The FIPS {{site.base_gateway}} package is FIPS 140-2 compliant. Compliance means that the software has met all of the rules of FIPS 140-2, but has not been submitted to a NIST testing lab for validation.
+
+
+{{site.ee_product_name}} provides a FIPS 140-2 compliant package for **Ubuntu 20.04** {% if_version gte:3.1.x %}, **Ubuntu 22.04**, and **Red Hat Enterprise 8** {% endif_version %}. This package provides compliance for the core {{site.base_gateway}} product{% if_version gte:3.1.x %} and all out of the box plugins{% endif_version %}.
+
+The package replaces the primary library in {{site.base_gateway}}, OpenSSL, with [BoringSSL](https://boringssl.googlesource.com/boringssl/), which at its core uses the FIPS 140-2 validated BoringCrypto for cryptographic operations.
+
+## FIPS implementation
+### Password hashing
+
+Password hashing is designed to protect the original password from being reversed. Although FIPS 140-2 doesn't mention password hashing, {{site.base_gateway}} uses key derivation functions that are approved:
```

Comment:
yes this doesn't seem relevant to our implementation.

Reply:
i'm not pretty sure what would be a best words here. functions like bcrypt doesn't really have a counterpart in fips world. what it tries to solve is different than message digest (which focuses on integrity and validation) or key derivation (which uses to generate key). there's no mention of password hashing in any fips document from my previous research, as well as some discussion i found online.

also do note although we pick a closest function that are included in fips world, the strength of pbkdf2 compared to bcrypt regarding password hashing is still relativelty weaker.

Reply:
we've also been vague on what kind of scope are we "fips-compliat" for. for example, fips-140-2 only defines the algorithms (and parameters) that are fips approved, but then it comes to SSL, which is defined in fips-186-4, and we can just follow only fips-140-2 and ignore SSL since most customer will need it and it's been asked even more than plugins.
and then key derivation function is in another document, under the hood, pbkdf2 is hmac+sha1 by default, both are fips approved algorithms.

https://docs.google.com/document/d/159ZOww97DbSr_elvoi1LgZ_Ljd9XKdB0MOzQnkyXtbc/edit#heading=h.ebkf1o96l282
this internal doc described the fips official publications we involved